icu4c/source/common/unicode/unistr.h

```diff
@@ -3140,21 +3134,20 @@ class U_COMMON_API UnicodeString : public Replaceable
    * on the compiler command line or similar.
    *
    * Note, for string literals:
-   * Since C++17 and ICU 76, you can use std::u16string_view literals with compile-time
+   * Since C++17 and ICU 76, you can use UTF-16 string literals with compile-time
    * length determination:
    * \code
-   * #include &lt;string_view&gt;
-   * using namespace std::string_view_literals;
-   * UnicodeString str(u"literal"sv);
-   * if (str == u"other literal"sv) { ... }
+   * UnicodeString str(u"literal");
+   * if (str == u"other literal") { ... }
    * \endcode
    *
    * @param text NUL-terminated UTF-16 string
    * @stable ICU 59
    */
   UNISTR_FROM_STRING_EXPLICIT UnicodeString(const wchar_t *text) :
-      UnicodeString(ConstChar16Ptr(text)) {}
+      UnicodeString(ConstChar16Ptr(text), -1) {}
 #endif
+#endif  // U_HIDE_DRAFT_API
```

Comment:
I just noticed that the from-nullptr constructor, which follows next, is not being removed. It's the only remaining version of the from-raw-pointer-only constructors.

Does the new toU16StringViewNullable not handle this case?

Reply:
I would be possible to use `toU16StringViewNullable()` to handle also `std::nullptr_t` but it's not necessary to do that so I thought it logical to not change more than necessary, it doesn't clash with anything and the specialized overload is possibly even ever so slightly faster than taking the detour through an empty string view.